### PR TITLE
Feat: Flashcard Streaks

### PR DESCRIPTION
**Pull request by jeremyalv**
# Overview
This Pull Request introduces the `Check Flashcard Streaks` and `Get My Streaks` feature for vlecture. The Streaks API tracks the review streak whenever the user reviews a Flashcard. Using Streaks, users can be held accountable and not  review their flashcards regularly. 

A streak continues if the user reviews it within 24 hours since last review, else, the streak is broken. There is a separate service for calculating how long are the`length_days` of the streak. 

# Major Features
- Streak Continuation/Termination Service
- Streak Length of Days Calculation Service
- Streak Fetching Service
- Streaks Data Model implementation
- TDD Approach & Unit Testing for Streaks API

# Minor Features
- Fixed bug causing dependency installation error on Ubuntu 22.04 (on Dockerfile)
- Add environment vars for new MongoDB collection name
- Swap ENV vars utility script -- use it like: `bash scripts/swap-env.sh dev` will switch to DEV env as the main ENV of project
- Create testing utils for Streaks API

# Endpoint Documentation
## Check Streak (Continue/Terminate)
**URL**: `/v1/streaks/check`

### Request
**Headers**:
Authorization: "Bearer `{{access_token}}`"

**Body**:
```
None
```

### Response
If a new Streak is created: Streaks object (`src.models.streaks > Streaks`)
Else: A single string response (FE can trust that the API will do the right thing on the background)

## View All Streaks
**URL**: `/v1/streaks/all`

### Request
**Headers**:
Authorization: "Bearer `{{access_token}}`"

**Body**:
```
None
```

### Response
A `List` of `Streaks` objects(`src.models.streaks > Streaks`)

### CODE REVIEW

**Comment by annavaws** [comment]
When a user reviews flashcards, does it mean they have to access the 'start test' on the flashcard set page, or does simply clicking on the flashcard set also count?

**Comment by jeremyalv** [comment]
> When a user reviews flashcards, does it mean they have to access the 'start test' on the flashcard set page, or does simply clicking on the flashcard set also count?

Users must click on Start Test to ensure they actually review the flashcards to be counted as a streak.

**Review by daffafaizan — APPROVED** [review]

File: src/controllers/streaks.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  streak_decision = service.determine_streak_decision(
+    current_time=current_time,
+    streak_last_updated=latest_streak.updated_at
+  )
+
+  if streak_decision == StreakDecisionEnum.ERROR:
+    return JSONResponse(
+      status_code=http.HTTPStatus.INTERNAL_SERVER_ERROR,
+      content="Error while determining streak decision."
+    )
```
Comment (annavaws) [inline src/controllers/streaks.py:91]: If an error occurs, does it affect all streaks uniformly, or are there different error handling mechanisms for specific streaks?

Comment (jeremyalv) [reply src/controllers/streaks.py:91]: The error only affects that specific streak, as each Decision is made on an individual streak.

**Comment by annavaws** [comment]
> > When a user reviews flashcards, does it mean they have to access the 'start test' on the flashcard set page, or does simply clicking on the flashcard set also count?
> 
> Users must click on Start Test to ensure they actually review the flashcards to be counted as a streak.

Sorry, i just realized i think it's when user click Finish Review to really ensure they actually done reviewing the flashcards?

**Pull request merged**